client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java

```diff
@@ -38,7 +41,10 @@ public void call() {
             eventListener.expectOnlyCreatedEvent(1, cache(cacheName));
          }
       });
-      detectThreadLeaks(".*Client-Listener-ClientListenerLeakTest-.*");
+      WaitFor
+              .condition(() -> listThreadsMatchingRegexp(".*Client-Listener-ClientListenerLeakTest-.*").size() == 0)
+              .atMost(10, TimeUnit.MILLISECONDS)
+              .await();
```

Comment:
@slaskawi, the `stop()` method already waits for 5 seconds, does a 10ms extra wait really make a difference? Or is the way we check for leaked threads different from the way `awaitTermination()` checks if the threads are done, and perhaps we should change that?